UI/DialogSystem/DialogSystem.cs

```diff
@@ -0,0 +1,71 @@
+using Godot;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+public class DialogSystem : CanvasLayer 
+{
+    private RichTextLabel _textLabel;
+    private IEnumerator<float> _coRoutine;
+    private Queue<string> _dialogQueue = new Queue<string>();
+
+    
+    
+    private float _coDelay = 0f;
+    private float _requestedCoDelay = 0f;
+    
+    
+    
+
+    [Export]
+    public float letterIntervalS = 0.05f;
+    
+
+
+    public override void _Ready()
+    {
+        _textLabel = GetNode<RichTextLabel>("TextContainer/MainText");
+        _coRoutine = InnerProcess().GetEnumerator();
+    }
+
+    public void PlayDialog(string message)
+    {    
+        _dialogQueue.Enqueue(message);
+    }
+
+    // The actual UI Co-routine, return value indicates desired delay between actions in seconds
+    private IEnumerable<float> InnerProcess()
+    {
+        while(true) {
```

Comment:
C# conventions, { on a new line.
Also, is the while (true) relly necessary ^^' i'm really not confident reading this kind of code...

Reply:
Yes it is. 
Its a c# Iterator block "https://www.c-sharpcorner.com/UploadFile/5ef30d/understanding-yield-return-in-C-Sharp/"
It autogenerates a statemachine, and allows me to have a method where i can say "run until next step"
Really usefull in long running background operations (Such as having text slowly being typed)